code401challenges/src/main/java/code401challenges/BinarySearch.java

```diff
@@ -6,19 +6,21 @@ public static void main(String[] args){
     System.out.println(binarySearch(inputArr, 15));
   }
 
-  public static int binarySearch(int[] sortedArr, int target){
-    int midIndex = sortedArr.length/2;
-    while(midIndex < sortedArr.length && midIndex > 0){
-      if(sortedArr[midIndex] == target){
-        return midIndex;
-      } else if(sortedArr[midIndex] < target){
-        midIndex = (midIndex + sortedArr.length)/2;
-      } else if(sortedArr[midIndex] > target){
-        midIndex = midIndex/2;
-      } else {
-        return -1;
+  public static int binarySearch(int[] arr, int target){
+    int minIndex = 0;
+    int maxIndex = arr.length - 1;
+    while(minIndex <= maxIndex ){
+      int guessIndex = (minIndex + maxIndex) / 2;
```

Comment:
Each time this loops `guessIndex` is declared and assigned a value. If you declare `guessIndex` outside the loop and just reassign mid inside the loop, your loop is doing a little less work. The efficiency of this is negligible but it's worth thinking about!